File: leetcode/src/easy/PathSum.java
package easy;

public class PathSum {

	public boolean hasPathSum(TreeNode root, int sum) {
		if (root == null) return false;
		
		if (root.val == sum && root.right == null && root.left == null) {
			return true;
		} else {
			return hasPathSum(root.left, sum - root.val) || hasPathSum(root.right, sum - root.val);
		}
    }
}

File: easy_py/PathSum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param {TreeNode} root
    # @param {integer} sum
    # @return {boolean}
    def hasPathSum(self, root, sum):
        if root == None:
            return False
        if root.val == sum and root.right == None and root.left == None:
            return True
        return self.hasPathSum(root.left, sum-root.val) or self.hasPathSum(root.right, sum-root.val)